functions/Get-DbaFirewallRule.ps1

```diff
@@ -0,0 +1,235 @@
+
+        As long as you can read this note here, there may be breaking changes in future versions.
+        So please review your scripts using this command after updating dbatools.
+
+    .PARAMETER SqlInstance
+        The target SQL Server instance or instances.
+
+    .PARAMETER Credential
+        Credential object used to connect to the Computer as a different user.
+
+    .PARAMETER Type
+        Returns firewall rules for the given type(s).
+        Valid values are:
+            Engine - for the SQL Server instance
+            Browser - for the SQL Server Browser
+            AllOnComputer - for all firewall rules on the target computer related to SQL Server
+        If this parameter is not used, the firewall rule for the SQL Server instance will be returned
+        and in case the instance is listening on a port other than 1433,
+        also the firewall rule for the SQL Server Browser will be returned.
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: Firewall, Network
+        Author: Andreas Jordan (@JordanOrdix), ordix.de
+
+        Website: https://dbatools.io
+        Copyright: (c) 2021 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Get-DbaFirewallRule
+
+    .EXAMPLE
+        PS C:\> Get-DbaFirewallRule -SqlInstance SRV1
+
+        Returns the firewall rule for the default instance on SRV1.
+        In case the instance is not listening on port 1433, it also returnes the firewall rule for the SQL Server Browser.
+
+    .EXAMPLE
+        PS C:\> Get-DbaFirewallRule -SqlInstance SRV1\SQL2016 -Type Engine
+
+        Returns only the firewall rule for the instance SQL2016 on SRV1.
+
+    .EXAMPLE
+        PS C:\> Get-DbaFirewallRule -SqlInstance SRV1\SQL2016 -Type Browser
+        PS C:\> Get-DbaFirewallRule -SqlInstance SRV1 -Type Browser
+
+        Both commands return the firewall rule for the SQL Serer Browser on SRV1.
+        As the Browser is not bound to a specific instance, only the computer part of SqlInstance is used.
+
+    .EXAMPLE
+        PS C:\> Get-DbaFirewallRule -SqlInstance SRV1\SQL2016 -Type AllOnComputer
+
+        Returns all firewall rules on the computer SRV1 related to SQL Server.
+        The value "AllOnComputer" only uses the computer name part of SqlInstance.
+
+    #>
+    [CmdletBinding()]
+    param (
+        [parameter(Mandatory, ValueFromPipeline)]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$Credential,
+        [ValidateSet('Engine', 'Browser', 'AllOnComputer')]
```

Comment:
Can we get AnalysisServices, IntegrationServices, ReportingServices here too?

If so, can we get the common things listed on `Install-DbaInstance -Feature`? 

```
"All", "Engine", "Tools", "Replication", "FullText", "DataQuality", "PolyBase", "MachineLearning", "AnalysisServices",
            "ReportingServices", "ReportingForSharepoint", "SharepointAddin", "IntegrationServices", "MasterDataServices", "PythonPackages", "RPackages",
            "ReplayController", "ReplayClient", "SDK", "BIDS", 
```

Reply:
Yes, if we find the time to code that.